producer.py

```diff
@@ -72,28 +72,56 @@ async def main():
         # Consider only last two releases by default
         adviser_versions = [str(v) for v in source.get_sorted_package_versions(package_name)][:number_releases]
 
+    justifications_collected: List[Dict[str, Any]] = []
+
+    for adviser_version in adviser_versions:
+        justifications_collected = Adviser.create_adviser_dataframe(
+            adviser_version=adviser_version,
+            adviser_files=adviser_files,
+            justifications_collected=justifications_collected,
+        )
+
+    adviser_dataframe = Adviser._create_adviser_dataframe(justifications_collected)
+
+    if adviser_dataframe.empty:
+        _LOGGER.exception("No adviser documents identified")
+        return
+
+    adviser_dataframe["date_"] = [
+        pd.to_datetime(str(v_date)).strftime("%Y-%m-%d") for v_date in adviser_dataframe["date"].values
+    ]
+
     for i in range(0, EVALUATION_METRICS_DAYS):
         date = datetime.datetime.utcnow() - datetime.timedelta(days=i)
         _LOGGER.info(f"Date considered: {date.strftime('%Y-%m-%d')}")
 
         advise_justifications: List[Dict[str, Any]] = []
 
         for adviser_version in adviser_versions:
-            adviser_dataframe = Adviser.create_adviser_dataframe(
-                adviser_version=adviser_version, adviser_files=adviser_files
-            )
-
-            adviser_summary_dataframe = Adviser.create_summary_dataframe(adviser_dataframe=adviser_dataframe)
-
-            for justification_type in ["INFO", "ERROR", "WARNING"]:
-
-                advise_justifications = parse_summary_dataframe(
-                    advise_justifications=advise_justifications,
-                    summary_dataframe=adviser_summary_dataframe,
-                    date_filter=date,
-                    justification_type=justification_type,
-                    adviser_version=adviser_version,
-                )
+            for unique_message in adviser_dataframe["message"].unique():
+                subset_df = adviser_dataframe[
+                    (adviser_dataframe["message"] == unique_message)
+                    & (adviser_dataframe["date_"] == str(date.strftime("%Y-%m-%d")))
+                    & (adviser_dataframe["analyzer_version"] == adviser_version)
+                ]
+
+                if not subset_df.empty:
+                    counts = subset_df.shape[0]
+
+                    types = [t for t in subset_df["type"].unique()]
+
+                    if len(types) > 1:
+                        _LOGGER.warning(f"type assigned to same message is different {types}")
```

Comment:
I bluntly remember that using formatted strings actually slows down the `logger` in some way, @fridex is that still right?
if so this should probably be
```suggestion
                        _LOGGER.warning("type assigned to same message is different %s", types)
```

Reply:
Thanks!